prow/plugins/trigger/trigger.go

```diff
@@ -241,6 +249,12 @@ func runOrSkipRequested(c client, pr *github.PullRequest, requestedJobs []config
 			toRun.Insert(job.Context)
 		} else if !job.SkipReport {
 			toSkip.Insert(job.Context)
+			// we need to post context statuses for all jobs; if a job is slated to
+			// run after the success of a parent job that is skipped, it must be
+			// skipped as well
+			for _, child := range job.RunAfterSuccess {
```

Comment:
Can this whole block now be replaced with a single `toSkip.Insert(allContexts(job.Context))`?